server/src/database/gameDB/base.ts

```diff
@@ -22,35 +23,61 @@ const gameComputer: Computer<BaseUnifiedGame, UnifiedGameComputed> = {
 type WatchableGame = WatchableResource<BaseUnifiedGame, UnifiedGameComputed>
 
 const gameDB: Record<string, WatchableGame> = {}
+const storage = new StoreFile<BaseUnifiedGame>('game', new RemoteStorage())
 
 export function gameInProgress (game: UnifiedGame): boolean {
   return game.gameStatus === 'Started' || game.gameStatus === 'Setup'
 }
 
-export function gameExists (gameId: string): boolean {
-  return !!gameDB[gameId]
+export async function gameExists (gameId: string): Promise<boolean> {
+  if (gameDB[gameId]) return true
+
+  const gameFromStorage = await storage.getFile(gameId)
+  if (gameFromStorage) {
+    gameDB[gameId] = new WatchableResource(gameFromStorage, gameComputer)
+    gameDB[gameId].subscribe((value) => {
+      storage.putFile(gameId, value as BaseUnifiedGame)
+        .catch((e) => { console.error(e) })
+    })
+    return true
+  }
+
+  return false
 }
 
-export function retrieveGame (gameId: string): WatchableGame {
-  if (!gameExists(gameId)) {
+export async function retrieveGame (gameId: string): Promise<WatchableGame> {
+  if (!(await gameExists(gameId))) {
     throw new Error(`${JSON.stringify(gameId)} not found`)
   }
 
   return gameDB[gameId]
 }
 
-export function getGame (gameId: string): UnifiedGame {
-  return retrieveGame(gameId).readOnce()
+export async function getGame (gameId: string): Promise<UnifiedGame> {
+  return (await retrieveGame(gameId)).readOnce()
 }
 
-export function addGame (gameId: string, game?: BaseUnifiedGame): boolean {
+export async function addGame (gameId: string): Promise<boolean> {
   console.log(`adding ${gameId}`)
-  if (gameExists(gameId)) {
+  if (await gameExists(gameId)) {
     throw new Error('Game already exists')
   }
 
-  gameDB[gameId] = new WatchableResource(game ?? createGame(), gameComputer)
-  addScript(gameId)
+  gameDB[gameId] = new WatchableResource(createGame(), gameComputer)
+  gameDB[gameId].subscribe((value) => {
+    storage.putFile(gameId, value as BaseUnifiedGame)
+      .catch((e) => { console.error(e) })
+  })
+  await addScript(gameId)
+
+  return true
+}
+
+export async function addTestGame (gameId: string, game: BaseUnifiedGame): Promise<boolean> {
```

Comment:
We can just kill test games, and write them to S3 as a way to test this feature

Reply:
Hrrrm, this gets a bit wonky and reveals an issue.  We likely want to prevent writes by default on localhost, since we don't have a local S3